isis/src/tgo/apps/tgocassisstitch/tgocassisstitch.cpp

```diff
@@ -66,20 +66,37 @@ namespace Isis {
       throw IException(e, IException::Unknown, msg, _FILEINFO_);
     }
 
+    // read the optional cubename
+    FileName frameletCubeFlag(ui.GetCubeName("CUBENAME"));
+    QString frameletCubeName = frameletCubeFlag.expanded();
+
     // Stitch together the individual frames
-    FileName outputFileName(ui.GetCubeName("OUTPUTPREFIX"));
-    QString outputBaseName = outputFileName.expanded();
+    FileName outputPrefix(ui.GetCubeName("OUTPUTPREFIX"));
+    FileName outputSuffix(ui.GetCubeName("OUTPUTSUFFIX"));
+    QString outputPrefBaseName = outputPrefix.expanded();
+    QString outputSuffBaseName = outputSuffix.expanded();
     QStringList frameKeys = frameMap.uniqueKeys();
     Progress stitchProgress;
     stitchProgress.SetText("Stitching Frames");
     stitchProgress.SetMaximumSteps(frameKeys.size());
     stitchProgress.CheckStatus();
+    
     foreach(QString frameKey, frameKeys) {
       try {
         QString frameIdentifier = frameKey.split("/").last();
-        FileName frameFileName(outputBaseName + "-" + frameIdentifier + ".cub");
-        stitchFrame( frameMap.values(frameKey), frameFileName );
-        stitchProgress.CheckStatus();
+        if (frameletCubeName != "") {
+          FileName frameFileName(outputPrefBaseName +
+                               frameletCubeName +
+                               outputSuffBaseName + ".cub");
+          stitchFrame( frameMap.values(frameKey), frameFileName );
+          stitchProgress.CheckStatus();
+        } else {
+          FileName frameFileName(outputPrefBaseName +
+                                 frameIdentifier +
+                                 outputSuffBaseName + ".cub");
+          stitchFrame( frameMap.values(frameKey), frameFileName );
+          stitchProgress.CheckStatus();
```

Comment:
Can we move this outside of the if block logic? You'll also need to moved the frameFileName declaration outside of the if block so it can be accessed outside of the logic block

Reply:
should be addressed with latest commit